<!DOCTYPE html>
<!-- saved from url=(0092)file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html -->
<html><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8">


<meta name="generator" content="pandoc">
<meta http-equiv="X-UA-Compatible" content="IE=EDGE">

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Roxana Andreea Moldovan Moldovan">

<meta name="date" content="2022-06-11">

<title>Tarea4. An√°lisis de grupos de genes</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




<script type="text/javascript" src="./tarea4_files/MathJax.js.descargar"></script><style type="text/css">.MathJax_Hover_Frame {border-radius: .25em; -webkit-border-radius: .25em; -moz-border-radius: .25em; -khtml-border-radius: .25em; box-shadow: 0px 0px 15px #83A; -webkit-box-shadow: 0px 0px 15px #83A; -moz-box-shadow: 0px 0px 15px #83A; -khtml-box-shadow: 0px 0px 15px #83A; border: 1px solid #A6D ! important; display: inline-block; position: absolute}
.MathJax_Menu_Button .MathJax_Hover_Arrow {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 4px; -webkit-border-radius: 4px; -moz-border-radius: 4px; -khtml-border-radius: 4px; font-family: 'Courier New',Courier; font-size: 9px; color: #F0F0F0}
.MathJax_Menu_Button .MathJax_Hover_Arrow span {display: block; background-color: #AAA; border: 1px solid; border-radius: 3px; line-height: 0; padding: 4px}
.MathJax_Hover_Arrow:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_Hover_Arrow:hover span {background-color: #CCC!important}
</style><style type="text/css">#MathJax_About {position: fixed; left: 50%; width: auto; text-align: center; border: 3px outset; padding: 1em 2em; background-color: #DDDDDD; color: black; cursor: default; font-family: message-box; font-size: 120%; font-style: normal; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; border-radius: 15px; -webkit-border-radius: 15px; -moz-border-radius: 15px; -khtml-border-radius: 15px; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_About.MathJax_MousePost {outline: none}
.MathJax_Menu {position: absolute; background-color: white; color: black; width: auto; padding: 2px; border: 1px solid #CCCCCC; margin: 0; cursor: default; font: menu; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; z-index: 201; box-shadow: 0px 10px 20px #808080; -webkit-box-shadow: 0px 10px 20px #808080; -moz-box-shadow: 0px 10px 20px #808080; -khtml-box-shadow: 0px 10px 20px #808080; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
.MathJax_MenuItem {padding: 2px 2em; background: transparent}
.MathJax_MenuArrow {position: absolute; right: .5em; padding-top: .25em; color: #666666; font-size: .75em}
.MathJax_MenuActive .MathJax_MenuArrow {color: white}
.MathJax_MenuArrow.RTL {left: .5em; right: auto}
.MathJax_MenuCheck {position: absolute; left: .7em}
.MathJax_MenuCheck.RTL {right: .7em; left: auto}
.MathJax_MenuRadioCheck {position: absolute; left: 1em}
.MathJax_MenuRadioCheck.RTL {right: 1em; left: auto}
.MathJax_MenuLabel {padding: 2px 2em 4px 1.33em; font-style: italic}
.MathJax_MenuRule {border-top: 1px solid #CCCCCC; margin: 4px 1px 0px}
.MathJax_MenuDisabled {color: GrayText}
.MathJax_MenuActive {background-color: Highlight; color: HighlightText}
.MathJax_MenuDisabled:focus, .MathJax_MenuLabel:focus {background-color: #E8E8E8}
.MathJax_ContextMenu:focus {outline: none}
.MathJax_ContextMenu .MathJax_MenuItem:focus {outline: none}
#MathJax_AboutClose {top: .2em; right: .2em}
.MathJax_Menu .MathJax_MenuClose {top: -10px; left: -10px}
.MathJax_MenuClose {position: absolute; cursor: pointer; display: inline-block; border: 2px solid #AAA; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; font-family: 'Courier New',Courier; font-size: 24px; color: #F0F0F0}
.MathJax_MenuClose span {display: block; background-color: #AAA; border: 1.5px solid; border-radius: 18px; -webkit-border-radius: 18px; -moz-border-radius: 18px; -khtml-border-radius: 18px; line-height: 0; padding: 8px 0 6px}
.MathJax_MenuClose:hover {color: white!important; border: 2px solid #CCC!important}
.MathJax_MenuClose:hover span {background-color: #CCC!important}
.MathJax_MenuClose:hover:focus {outline: none}
</style><style type="text/css">.MathJax_Preview .MJXf-math {color: inherit!important}
</style><style type="text/css">.MJX_Assistive_MathML {position: absolute!important; top: 0; left: 0; clip: rect(1px, 1px, 1px, 1px); padding: 1px 0 0 0!important; border: 0!important; height: 1px!important; width: 1px!important; overflow: hidden!important; display: block!important; -webkit-touch-callout: none; -webkit-user-select: none; -khtml-user-select: none; -moz-user-select: none; -ms-user-select: none; user-select: none}
.MJX_Assistive_MathML.MJX_Assistive_MathML_Block {width: 100%!important}
</style><style type="text/css">#MathJax_Zoom {position: absolute; background-color: #F0F0F0; overflow: auto; display: block; z-index: 301; padding: .5em; border: 1px solid black; margin: 0; font-weight: normal; font-style: normal; text-align: left; text-indent: 0; text-transform: none; line-height: normal; letter-spacing: normal; word-spacing: normal; word-wrap: normal; white-space: nowrap; float: none; -webkit-box-sizing: content-box; -moz-box-sizing: content-box; box-sizing: content-box; box-shadow: 5px 5px 15px #AAAAAA; -webkit-box-shadow: 5px 5px 15px #AAAAAA; -moz-box-shadow: 5px 5px 15px #AAAAAA; -khtml-box-shadow: 5px 5px 15px #AAAAAA; filter: progid:DXImageTransform.Microsoft.dropshadow(OffX=2, OffY=2, Color='gray', Positive='true')}
#MathJax_ZoomOverlay {position: absolute; left: 0; top: 0; z-index: 300; display: inline-block; width: 100%; height: 100%; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
#MathJax_ZoomFrame {position: relative; display: inline-block; height: 0; width: 0}
#MathJax_ZoomEventTrap {position: absolute; left: 0; top: 0; z-index: 302; display: inline-block; border: 0; padding: 0; margin: 0; background-color: white; opacity: 0; filter: alpha(opacity=0)}
</style><style type="text/css">.MathJax_Preview {color: #888}
#MathJax_Message {position: fixed; left: 1em; bottom: 1.5em; background-color: #E6E6E6; border: 1px solid #959595; margin: 0px; padding: 2px 8px; z-index: 102; color: black; font-size: 80%; width: auto; white-space: nowrap}
#MathJax_MSIE_Frame {position: absolute; top: 0; left: 0; width: 0px; z-index: 101; border: 0px; margin: 0px; padding: 0px}
.MathJax_Error {color: #CC0000; font-style: italic}
</style><style type="text/css">.MJXp-script {font-size: .8em}
.MJXp-right {-webkit-transform-origin: right; -moz-transform-origin: right; -ms-transform-origin: right; -o-transform-origin: right; transform-origin: right}
.MJXp-bold {font-weight: bold}
.MJXp-italic {font-style: italic}
.MJXp-scr {font-family: MathJax_Script,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-frak {font-family: MathJax_Fraktur,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-sf {font-family: MathJax_SansSerif,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-cal {font-family: MathJax_Caligraphic,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-mono {font-family: MathJax_Typewriter,'Times New Roman',Times,STIXGeneral,serif}
.MJXp-largeop {font-size: 150%}
.MJXp-largeop.MJXp-int {vertical-align: -.2em}
.MJXp-math {display: inline-block; line-height: 1.2; text-indent: 0; font-family: 'Times New Roman',Times,STIXGeneral,serif; white-space: nowrap; border-collapse: collapse}
.MJXp-display {display: block; text-align: center; margin: 1em 0}
.MJXp-math span {display: inline-block}
.MJXp-box {display: block!important; text-align: center}
.MJXp-box:after {content: " "}
.MJXp-rule {display: block!important; margin-top: .1em}
.MJXp-char {display: block!important}
.MJXp-mo {margin: 0 .15em}
.MJXp-mfrac {margin: 0 .125em; vertical-align: .25em}
.MJXp-denom {display: inline-table!important; width: 100%}
.MJXp-denom > * {display: table-row!important}
.MJXp-surd {vertical-align: top}
.MJXp-surd > * {display: block!important}
.MJXp-script-box > *  {display: table!important; height: 50%}
.MJXp-script-box > * > * {display: table-cell!important; vertical-align: top}
.MJXp-script-box > *:last-child > * {vertical-align: bottom}
.MJXp-script-box > * > * > * {display: block!important}
.MJXp-mphantom {visibility: hidden}
.MJXp-munderover {display: inline-table!important}
.MJXp-over {display: inline-block!important; text-align: center}
.MJXp-over > * {display: block!important}
.MJXp-munderover > * {display: table-row!important}
.MJXp-mtable {vertical-align: .25em; margin: 0 .125em}
.MJXp-mtable > * {display: inline-table!important; vertical-align: middle}
.MJXp-mtr {display: table-row!important}
.MJXp-mtd {display: table-cell!important; text-align: center; padding: .5em 0 0 .5em}
.MJXp-mtr > .MJXp-mtd:first-child {padding-left: 0}
.MJXp-mtr:first-child > .MJXp-mtd {padding-top: 0}
.MJXp-mlabeledtr {display: table-row!important}
.MJXp-mlabeledtr > .MJXp-mtd:first-child {padding-left: 0}
.MJXp-mlabeledtr:first-child > .MJXp-mtd {padding-top: 0}
.MJXp-merror {background-color: #FFFF88; color: #CC0000; border: 1px solid #CC0000; padding: 1px 3px; font-style: normal; font-size: 90%}
.MJXp-scale0 {-webkit-transform: scaleX(.0); -moz-transform: scaleX(.0); -ms-transform: scaleX(.0); -o-transform: scaleX(.0); transform: scaleX(.0)}
.MJXp-scale1 {-webkit-transform: scaleX(.1); -moz-transform: scaleX(.1); -ms-transform: scaleX(.1); -o-transform: scaleX(.1); transform: scaleX(.1)}
.MJXp-scale2 {-webkit-transform: scaleX(.2); -moz-transform: scaleX(.2); -ms-transform: scaleX(.2); -o-transform: scaleX(.2); transform: scaleX(.2)}
.MJXp-scale3 {-webkit-transform: scaleX(.3); -moz-transform: scaleX(.3); -ms-transform: scaleX(.3); -o-transform: scaleX(.3); transform: scaleX(.3)}
.MJXp-scale4 {-webkit-transform: scaleX(.4); -moz-transform: scaleX(.4); -ms-transform: scaleX(.4); -o-transform: scaleX(.4); transform: scaleX(.4)}
.MJXp-scale5 {-webkit-transform: scaleX(.5); -moz-transform: scaleX(.5); -ms-transform: scaleX(.5); -o-transform: scaleX(.5); transform: scaleX(.5)}
.MJXp-scale6 {-webkit-transform: scaleX(.6); -moz-transform: scaleX(.6); -ms-transform: scaleX(.6); -o-transform: scaleX(.6); transform: scaleX(.6)}
.MJXp-scale7 {-webkit-transform: scaleX(.7); -moz-transform: scaleX(.7); -ms-transform: scaleX(.7); -o-transform: scaleX(.7); transform: scaleX(.7)}
.MJXp-scale8 {-webkit-transform: scaleX(.8); -moz-transform: scaleX(.8); -ms-transform: scaleX(.8); -o-transform: scaleX(.8); transform: scaleX(.8)}
.MJXp-scale9 {-webkit-transform: scaleX(.9); -moz-transform: scaleX(.9); -ms-transform: scaleX(.9); -o-transform: scaleX(.9); transform: scaleX(.9)}
.MathJax_PHTML .noError {vertical-align: ; font-size: 90%; text-align: left; color: black; padding: 1px 3px; border: 1px solid}
</style></head>

<body><div id="MathJax_Message" style="display: none;"></div>




<h1 class="title toc-ignore">Tarea4. An√°lisis de grupos de genes</h1>
<h4 class="author">Roxana Andreea Moldovan Moldovan</h4>
<h4 class="date">2022-06-11</h4>



<hr>
<div id="introducci√≥n" class="section level1">
<h1>1. Introducci√≥n</h1>
<p>En esta tarea, partiendo de los datos de microarrays provenientes de
la <strong>tarea1</strong>, llevaremos a cabo distintos an√°lisis de
grupos de genes <strong>Go</strong> y <strong>KEGG</strong>.</p>
<p>&nbsp;</p>
<hr>
</div>
<div id="carga-de-datos." class="section level1">
<h1>2. Carga de datos.</h1>
<div id="paquetes-empleados" class="section level2">
<h2>Paquetes empleados:</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb1-1" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(pbapply)</span>
<span id="cb1-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb1-2" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(SummarizedExperiment,EnrichmentBrowser,GSEABase, tami, GSA)</span>
<span id="cb1-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb1-3" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(ReportingTools)</span>
<span id="cb1-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb1-4" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(pd.cyngene.<span class="dv">1</span>.<span class="fl">0.</span>st,org.Mmu.eg.db, GO.db)</span></code></pre></div>
</div>
<div id="carga-de-datos-del-experimento." class="section level2">
<h2>Carga de datos del experimento.</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb2-1" aria-hidden="true" tabindex="-1"></a>datos<span class="ot">=</span><span class="st">"geod75759_rma.rda"</span></span>
<span id="cb2-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">paste</span>(dir_datos, datos, <span class="at">sep=</span><span class="st">""</span>))</span></code></pre></div>
<p>&nbsp;</p>
<hr>
</div>
</div>
<div id="obtenci√≥n-de-grupos-de-genes" class="section level1">
<h1>3. Obtenci√≥n de grupos de genes:</h1>
<div id="grupos-de-genes-go" class="section level2">
<h2>3.1. Grupos de genes GO</h2>
<p>En este apartado vamos a hacer una an√°lisis de sobre representaci√≥n
con los grupos de genes de <strong>Go (Gene Ontology)</strong>.</p>
<div id="obtenci√≥n-del-geneset" class="section level3">
<h3>Obtenci√≥n del GeneSet</h3>
<p>Debido a que el estudio no presentaba una base de datos de anotaci√≥n
adecuada asiciada a nuestra sonda no podemos generar el
GeneSetCollection directamente, por ello obtenemos los grupos de genes y
los identificadores <strong>ENTREZID</strong> asociados a estos a estos
a partir de <strong>org.Mmu.eg.db</strong>. A continuaci√≥n, a partir de
los identificadores Go, la evidencia y el identificador ENTREZID, usando
las funciones <strong>GOFrame</strong> y <strong>GOAllFrame</strong>, e
indicando el organismo, anotamos la informaci√≥n.</p>
<p>Finalmente usamos <strong>GeneSetCollection</strong> para generar
colecciones de genes.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Tomamos la informaci√≥n de Go:</span></span>
<span id="cb3-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb3-2" aria-hidden="true" tabindex="-1"></a>frame <span class="ot">&lt;-</span> <span class="fu">toTable</span>(org.Mmu.egGO2EG) </span>
<span id="cb3-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Creamos un data frame con el identificador Go, la evidencia y el id del gen:</span></span>
<span id="cb3-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb3-4" aria-hidden="true" tabindex="-1"></a>goframeData <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(frame<span class="sc">$</span>go_id, frame<span class="sc">$</span>Evidence, frame<span class="sc">$</span>gene_id)</span>
<span id="cb3-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb3-5" aria-hidden="true" tabindex="-1"></a>goFrame <span class="ot">=</span> AnnotationDbi<span class="sc">::</span><span class="fu">GOFrame</span>(goframeData, <span class="at">organism =</span> <span class="st">"Macaca mulatta"</span>)</span>
<span id="cb3-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb3-6" aria-hidden="true" tabindex="-1"></a>goAllFrame <span class="ot">=</span> AnnotationDbi<span class="sc">::</span><span class="fu">GOAllFrame</span>(goFrame)</span>
<span id="cb3-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb3-7" aria-hidden="true" tabindex="-1"></a>mccGOgsc_all <span class="ot">=</span> GSEABase<span class="sc">::</span><span class="fu">GeneSetCollection</span>(goAllFrame, <span class="at">setType =</span> <span class="fu">GOCollection</span>())</span>
<span id="cb3-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb3-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Guardamos la colecci√≥n Go con todos los genes de Macaca mulatta:</span></span>
<span id="cb3-10"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mccGOgsc_all, <span class="at">file =</span> <span class="fu">paste0</span>(dir_datos,<span class="st">"mccGOgsc_all.rda"</span>))</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cargamos la colecci√≥n de Go:</span></span>
<span id="cb4-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">paste0</span>(dir_datos,<span class="st">"mccGOgsc_all.rda"</span>))</span></code></pre></div>
<p>En la colecci√≥n tenemos todos los identificadores asociados a cada
grupo Go, nos quedamos √∫nicamente con los que est√©n en nuestro
ExpressionSet.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Tomamos los que tienen correspondencia con nuestros identificadores:</span></span>
<span id="cb5-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="co">#Funci√≥n subsettingGeneSet:</span></span>
<span id="cb5-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-4" aria-hidden="true" tabindex="-1"></a>SelectSubGeneSet <span class="ot">&lt;-</span> <span class="cf">function</span>(GS, G_ID){ </span>
<span id="cb5-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-5" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Introducimos el GeneSet y los ENTREZID</span></span>
<span id="cb5-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geneIds</span>(GS) <span class="ot">=</span> <span class="fu">geneIds</span>(GS)[<span class="fu">is.element</span>( <span class="fu">geneIds</span>(GS), G_ID)]</span>
<span id="cb5-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-7" aria-hidden="true" tabindex="-1"></a>  GS</span>
<span id="cb5-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb5-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-10"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Grupos de genes Go con nuestros identificadores:</span></span>
<span id="cb5-11"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-11" aria-hidden="true" tabindex="-1"></a>mccGOgsc_reducido <span class="ot">&lt;-</span> <span class="fu">pblapply</span>(mccGOgsc_all, SelectSubGeneSet,</span>
<span id="cb5-12"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-12" aria-hidden="true" tabindex="-1"></a>                       <span class="at">G_ID=</span><span class="fu">fData</span>(geod75759_rma)<span class="sc">$</span>ENTREZID)</span>
<span id="cb5-13"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb5-13" aria-hidden="true" tabindex="-1"></a>mccGOgsc <span class="ot">&lt;-</span> <span class="fu">GeneSetCollection</span>(mccGOgsc_reducido)</span></code></pre></div>
<p>Observamos que tenemos el mismo n√∫mero de grupos pero hemos eliminado
los identificadores de genes que no se encuentran en nuestro
experimento.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">"Tenemos "</span>, <span class="fu">length</span>(mccGOgsc_all), <span class="st">" grupos segun la clsificaci√≥n GO."</span>)</span></code></pre></div>
<pre><code>## [1] "Tenemos 17472 grupos segun la clsificaci√≥n GO."</code></pre>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">"El primer grupo contiene "</span>, <span class="fu">length</span>(<span class="fu">geneIds</span>(mccGOgsc_all[[<span class="dv">1</span>]])), <span class="st">" genes."</span>)</span></code></pre></div>
<pre><code>## [1] "El primer grupo contiene 12 genes."</code></pre>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Una vez eliminados:</span></span>
<span id="cb10-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">"Tenemos "</span>, <span class="fu">length</span>(mccGOgsc), <span class="st">" grupos segun la clsificaci√≥n GO."</span>)</span></code></pre></div>
<pre><code>## [1] "Tenemos 17472 grupos segun la clsificaci√≥n GO."</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">"El primer grupo contiene "</span>, <span class="fu">length</span>(<span class="fu">geneIds</span>(mccGOgsc[[<span class="dv">1</span>]])), <span class="st">" genes."</span>)</span></code></pre></div>
<pre><code>## [1] "El primer grupo contiene 7 genes."</code></pre>
<p>En cada grupo tenemos un numero de genes, a continuaci√≥n obtendremos
informaci√≥n generalizado sobre el n√∫mero de genes que componen los
grupos.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb14-1" aria-hidden="true" tabindex="-1"></a>count_Go <span class="ot">&lt;-</span> <span class="fu">pbsapply</span>(<span class="fu">geneIds</span>(mccGOgsc), length)</span>
<span id="cb14-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(count_Go)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    0.00    1.00    3.00   33.48   10.00 6834.00</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb16-1" aria-hidden="true" tabindex="-1"></a>a <span class="ot">=</span> mccGOgsc[(<span class="fu">lapply</span>(<span class="fu">geneIds</span>(mccGOgsc), length) <span class="sc">&lt;=</span> <span class="dv">1</span>)]</span>
<span id="cb16-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(<span class="fu">length</span>(a), <span class="st">" grupos GO presenta 0 o un √∫nico gen."</span>)</span></code></pre></div>
<pre><code>## [1] "6144 grupos GO presenta 0 o un √∫nico gen."</code></pre>
<p>Nos quedaremos √∫nicamente con los grupos que presentan al menor dos
genes.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Tomamos los grupos con m√°s de un gen</span></span>
<span id="cb18-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb18-2" aria-hidden="true" tabindex="-1"></a>mccGOgsc <span class="ot">=</span> mccGOgsc[(<span class="fu">lapply</span>(<span class="fu">geneIds</span>(mccGOgsc), length) <span class="sc">&gt;</span> <span class="dv">1</span>)]</span>
<span id="cb18-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">"Al final nos quedamos con "</span>, <span class="fu">length</span>(mccGOgsc), <span class="st">"grupos Go"</span>)</span></code></pre></div>
<pre><code>## [1] "Al final nos quedamos con 11328grupos Go"</code></pre>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb20-1" aria-hidden="true" tabindex="-1"></a>count_Go <span class="ot">&lt;-</span> <span class="fu">pbsapply</span>(<span class="fu">geneIds</span>(mccGOgsc), length)</span>
<span id="cb20-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(count_Go)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    2.00    3.00    6.00   51.23   21.00 6834.00</code></pre>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">length</span>(mccGOgsc)</span></code></pre></div>
<pre><code>## [1] 11328</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Guardamos el conjunto:</span></span>
<span id="cb24-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb24-2" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mccGOgsc, <span class="at">file =</span> <span class="fu">paste0</span>(dir_datos,<span class="st">"mccGOgsc.rda"</span>))</span></code></pre></div>
</div>
</div>
<div id="grupos-de-genes-kegg." class="section level2">
<h2>3.2. Grupos de genes KEGG.</h2>
<p>En este apartado vamos a hacer una an√°lisis de sobre representaci√≥n
con los grupos de genes de <strong>KEGG (Kyoto Encyclopedia of Genes and
Genomes)</strong>.</p>
<div id="obtenci√≥n-del-genesetcollection" class="section level3">
<h3>Obtenci√≥n del GeneSetCollection</h3>
<p>En primer lugar debemos descargar los datos referentes a los grupos
de genes usando la funci√≥n <strong>getGenesets</strong> del paquete
<strong>EnrichmentBrowser</strong> y especificando:</p>
<ul>
<li><p>El organismo espec√≠fico usando su c√≥digo espec√≠fico de 3 letras,
en este caso <em>Macaca mulatta</em> (<strong>mcc</strong>).</p></li>
<li><p>La base da datos espec√≠fica, en este caso
<strong>kegg.</strong></p></li>
</ul>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Obtenemos la coleccion de genes de KEGG:</span></span>
<span id="cb25-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb25-2" aria-hidden="true" tabindex="-1"></a>mccKEGGgsc_all <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">getGenesets</span>(<span class="at">org =</span> <span class="st">"mcc"</span>, <span class="at">db =</span><span class="st">"kegg"</span>, </span>
<span id="cb25-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb25-3" aria-hidden="true" tabindex="-1"></a>                                                <span class="at">gene.id.type =</span> <span class="st">"ENTREZID"</span>,</span>
<span id="cb25-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb25-4" aria-hidden="true" tabindex="-1"></a>                                            <span class="at">return.type =</span> <span class="st">"GeneSetCollection"</span>)</span>
<span id="cb25-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb25-5" aria-hidden="true" tabindex="-1"></a>mccKEGGgsc_all</span>
<span id="cb25-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb25-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb25-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Guardamos los datos:</span></span>
<span id="cb25-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mccKEGGgsc_all, <span class="at">file =</span> (<span class="fu">paste0</span>(dir_datos, <span class="st">"mccKEGGgsc_all.rda"</span>)))</span></code></pre></div>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Cargamos los datos:</span></span>
<span id="cb26-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">paste0</span>(dir_datos, <span class="st">"mccKEGGgsc_all.rda"</span>))</span></code></pre></div>
<p>Nos quedamos √∫nicamente con los identificadores presentes en nuestro
experimento.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Funci√≥n subsettingGeneSet</span></span>
<span id="cb27-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-2" aria-hidden="true" tabindex="-1"></a>SelectSubGeneSet <span class="ot">&lt;-</span> <span class="cf">function</span>(GS, G_ID){ </span>
<span id="cb27-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Introducimos el GeneSet y los ENTREZID</span></span>
<span id="cb27-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geneIds</span>(GS) <span class="ot">=</span> <span class="fu">geneIds</span>(GS)[<span class="fu">is.element</span>( <span class="fu">geneIds</span>(GS), G_ID)]</span>
<span id="cb27-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-5" aria-hidden="true" tabindex="-1"></a>  GS</span>
<span id="cb27-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Grupos de genes KEGG con nuestros identificadores:</span></span>
<span id="cb27-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-9" aria-hidden="true" tabindex="-1"></a>mccKEGGgsc_reducido <span class="ot">&lt;-</span> <span class="fu">pblapply</span>(mccKEGGgsc_all, SelectSubGeneSet,</span>
<span id="cb27-10"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">G_ID=</span><span class="fu">fData</span>(geod75759_rma)<span class="sc">$</span>ENTREZID)</span>
<span id="cb27-11"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-11" aria-hidden="true" tabindex="-1"></a>mccKEGGgsc <span class="ot">&lt;-</span> <span class="fu">GeneSetCollection</span>(mccKEGGgsc_reducido)</span>
<span id="cb27-12"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Guardamos los datos:</span></span>
<span id="cb27-13"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb27-13" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mccKEGGgsc, <span class="at">file =</span> (<span class="fu">paste0</span>(dir_datos, <span class="st">"mccKEGGgsc.rda"</span>)))</span></code></pre></div>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">"Tenemos "</span>, <span class="fu">length</span>(mccKEGGgsc_all), <span class="st">" grupos segun la clsificaci√≥n KEGG."</span>)</span></code></pre></div>
<pre><code>## [1] "Tenemos 343 grupos segun la clsificaci√≥n KEGG."</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">"El primer grupo contiene "</span>, <span class="fu">length</span>(<span class="fu">geneIds</span>(mccKEGGgsc_all[[<span class="dv">1</span>]])), <span class="st">" genes."</span>)</span></code></pre></div>
<pre><code>## [1] "El primer grupo contiene 80 genes."</code></pre>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Una vez eliminados:</span></span>
<span id="cb32-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">"Tenemos "</span>, <span class="fu">length</span>(mccKEGGgsc), <span class="st">" grupos segun la clsificaci√≥n KEGG."</span>)</span></code></pre></div>
<pre><code>## [1] "Tenemos 343 grupos segun la clsificaci√≥n KEGG."</code></pre>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb34-1" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(<span class="st">"El primer grupo contiene "</span>, <span class="fu">length</span>(<span class="fu">geneIds</span>(mccKEGGgsc[[<span class="dv">1</span>]])), <span class="st">" genes."</span>)</span></code></pre></div>
<pre><code>## [1] "El primer grupo contiene 54 genes."</code></pre>
<p>Encontramos 343 grupos diferentes de genes seg√∫n la ruta en la que
participan.<br>
Algunos de estos son:</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">names</span>(mccKEGGgsc))</span></code></pre></div>
<pre><code>## [1] "mcc00010" "mcc00020" "mcc00030" "mcc00040" "mcc00051" "mcc00052"</code></pre>
<p>En cada grupo tenemos un n√∫mero de genes, a continuaci√≥n obtendremos
informaci√≥n generalizado sobre el n√∫mero de genes que componen los
grupos KEGG:</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb38-1" aria-hidden="true" tabindex="-1"></a>count_Go <span class="ot">&lt;-</span> <span class="fu">pbsapply</span>(<span class="fu">geneIds</span>(mccKEGGgsc), length)</span>
<span id="cb38-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(count_Go)</span></code></pre></div>
<pre><code>##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
##    3.00   29.00   54.00   69.13   88.00 1071.00</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb40-1" aria-hidden="true" tabindex="-1"></a>a <span class="ot">=</span> mccKEGGgsc[(<span class="fu">lapply</span>(<span class="fu">geneIds</span>(mccKEGGgsc), length) <span class="sc">&lt;=</span> <span class="dv">1</span>)]</span>
<span id="cb40-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">paste0</span>(<span class="fu">length</span>(a), <span class="st">" grupos KEGG presenta menos de 3 genes."</span>)</span></code></pre></div>
<pre><code>## [1] "0 grupos KEGG presenta menos de 3 genes."</code></pre>
<p>Por tanto nos quedamos con todos los grupos.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Guardamos los datos:</span></span>
<span id="cb42-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(mccKEGGgsc, <span class="at">file =</span> (<span class="fu">paste0</span>(dir_datos, <span class="st">"mccKEGGgsc.rda"</span>)))</span>
<span id="cb42-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">paste0</span>(dir_datos, <span class="st">"mccKEGGgsc.rda"</span>))</span></code></pre></div>
<p>&nbsp;</p>
<hr>
</div>
</div>
</div>
<div id="obtenci√≥n-del-summarized-experiment." class="section level1">
<h1>4. Obtenci√≥n del Summarized Experiment.</h1>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(geod75759_rma)</span></code></pre></div>
<pre><code>## [1] "ExpressionSet"
## attr(,"package")
## [1] "Biobase"</code></pre>
<p>En los apartados anteriores hemos trabajado con el
<strong>ExpressionSet</strong> de nuestros datos, ahora debemos
trasformar el <strong>ExpressionSet</strong> en un
<strong>SummarizedExperiment</strong>.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb45-1" aria-hidden="true" tabindex="-1"></a>se75759 <span class="ot">&lt;-</span> <span class="fu">makeSummarizedExperimentFromExpressionSet</span>(geod75759_rma)</span>
<span id="cb45-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb45-2" aria-hidden="true" tabindex="-1"></a>se75759</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 13242 18 
## metadata(3): experimentData annotation protocolData
## assays(1): exprs
## rownames(13242): 13579638 13579671 ... 14215578 14215636
## rowData names(5): ENTREZID PROBEID ENSEMBLTRANS GENSYMBOL ENSAMBL
## colnames(18): GSM1966868_7851_ExpWBC.CEL GSM1966867_4E78_Exp_WBC.CEL
##   ... GSM1966852_020A_Air_Nose.CEL GSM1966851_7D77_Air_Nose.CEL
## colData names(39): Source.Name Sample_type ...
##   Comment..Derived.ArrayExpress.FTP.file. FactorValue..organism.part.</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rowData</span>(se75759), <span class="dv">2</span>)</span></code></pre></div>
<pre><code>## DataFrame with 2 rows and 5 columns
##             ENTREZID   PROBEID       ENSEMBLTRANS   GENSYMBOL
##          &lt;character&gt; &lt;numeric&gt;        &lt;character&gt; &lt;character&gt;
## 13579638      699677  13579638 ENSMMUT00000032498      LRPAP1
## 13579671      720098  13579671 ENSMMUT00000011722   LOC720098
##                     ENSAMBL
##                 &lt;character&gt;
## 13579638 ENSMMUG00000023099
## 13579671 ENSMMUG00000008383</code></pre>
<p>Como podemos observar tenemos los identificadores a nivel de sonda
(<strong>PROBEID</strong>), pero nos interesa pasar de expresi√≥n a nivel
de sonda a expresi√≥n a nivel de gen (<strong>ENTREZID</strong>), para
ello usamos la funci√≥n <em><strong>probe2gene</strong></em>.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb49-1" aria-hidden="true" tabindex="-1"></a>se75759 <span class="ot">=</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">probe2gene</span>(se75759, <span class="at">from =</span> <span class="st">"PROBEID"</span>,</span>
<span id="cb49-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb49-2" aria-hidden="true" tabindex="-1"></a>                                        <span class="at">to =</span> <span class="st">"ENTREZID"</span>)</span>
<span id="cb49-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb49-3" aria-hidden="true" tabindex="-1"></a>se75759</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 13242 18 
## metadata(3): experimentData annotation protocolData
## assays(1): exprs
## rownames(13242): 699677 720098 ... 701102 707303
## rowData names(0):
## colnames(18): GSM1966868_7851_ExpWBC.CEL GSM1966867_4E78_Exp_WBC.CEL
##   ... GSM1966852_020A_Air_Nose.CEL GSM1966851_7D77_Air_Nose.CEL
## colData names(39): Source.Name Sample_type ...
##   Comment..Derived.ArrayExpress.FTP.file. FactorValue..organism.part.</code></pre>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(se75759)</span></code></pre></div>
<pre><code>## DataFrame with 13242 rows and 0 columns</code></pre>
<p>A√±adimos una variable de 0 y 1 llamada <strong>GROUP</strong> que
describa la segregaci√≥n en grupos seg√∫n el tratamiento.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb53-1" aria-hidden="true" tabindex="-1"></a>tratamiento <span class="ot">&lt;-</span> <span class="fu">colData</span>(se75759)<span class="sc">$</span>Sample_treatment</span>
<span id="cb53-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb53-2" aria-hidden="true" tabindex="-1"></a>tratamiento</span></code></pre></div>
<pre><code>##  [1] Formaldehyde Formaldehyde Formaldehyde Formaldehyde Formaldehyde
##  [6] Filtered_Air Filtered_Air Filtered_Air Filtered_Air Filtered_Air
## [11] Formaldehyde Formaldehyde Formaldehyde Formaldehyde Filtered_Air
## [16] Filtered_Air Filtered_Air Filtered_Air
## Levels: Formaldehyde Filtered_Air</code></pre>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb55-1" aria-hidden="true" tabindex="-1"></a>grupos <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(tratamiento <span class="sc">==</span> <span class="st">"Formaldehyde"</span>, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb55-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb55-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(se75759) <span class="ot">&lt;-</span> <span class="fu">cbind</span>(<span class="fu">colData</span>(se75759), <span class="at">GROUP=</span>grupos)</span>
<span id="cb55-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb55-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colData</span>(se75759),<span class="dv">1</span>)</span></code></pre></div>
<pre><code>## DataFrame with 1 row and 40 columns
##                             Source.Name Sample_type Sample_treatment
##                             &lt;character&gt;    &lt;factor&gt;         &lt;factor&gt;
## GSM1966868_7851_ExpWBC.CEL GSM1966868 1         WBC     Formaldehyde
##                            Comment..Sample_description.
##                                             &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL       Gene expression data..
##                            Comment..Sample_source_name. Comment..Sample_title.
##                                             &lt;character&gt;            &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL       white blood cells fr..  WBC_Formaldehyde_Rep5
##                            Characteristics..organism. Term.Source.REF
##                                           &lt;character&gt;     &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL        Macaca fascicularis             EFO
##                             Term.Accession.Number
##                                       &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL http://purl.obolibra..
##                            Characteristics..organism.part.
##                                                &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL          circulating white bl..
##                            Characteristics..sex. Term.Source.REF.1
##                                      &lt;character&gt;       &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL                  male               EFO
##                            Term.Accession.Number.1 Protocol.REF
##                                        &lt;character&gt;  &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL  http://purl.obolibra.. P-GSE75759-2
##                            Term.Source.REF.2 Protocol.REF.1 Term.Source.REF.3
##                                  &lt;character&gt;    &lt;character&gt;       &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL      ArrayExpress   P-GSE75759-3      ArrayExpress
##                                    Extract.Name Material.Type Protocol.REF.2
##                                     &lt;character&gt;   &lt;character&gt;    &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL GSM1966868 extract 1     total RNA   P-GSE75759-4
##                            Term.Source.REF.4 Labeled.Extract.Name
##                                  &lt;character&gt;          &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL      ArrayExpress      GSM1966868 LE 1
##                                             Label Protocol.REF.3
##                                       &lt;character&gt;    &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL terminal deoxynucleo..   P-GSE75759-5
##                            Term.Source.REF.5  Assay.Name Array.Design.REF
##                                  &lt;character&gt; &lt;character&gt;      &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL      ArrayExpress  GSM1966868     A-GEOD-21213
##                            Term.Source.REF.6 Technology.Type Protocol.REF.4
##                                  &lt;character&gt;     &lt;character&gt;    &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL      ArrayExpress     array assay   P-GSE75759-6
##                            Term.Source.REF.7        Array.Data.File
##                                  &lt;character&gt;            &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL      ArrayExpress GSM1966868_7851_ExpW..
##                            Comment..ArrayExpress.FTP.file. Protocol.REF.5
##                                                &lt;character&gt;    &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL          ftp://ftp.ebi.ac.uk/..   P-GSE75759-1
##                            Term.Source.REF.8     Normalization.Name
##                                  &lt;character&gt;            &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL      ArrayExpress GSM1966868_sample_ta..
##                            Derived.Array.Data.File
##                                        &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL  GSM1966868_sample_ta..
##                            Comment..Derived.ArrayExpress.FTP.file.
##                                                        &lt;character&gt;
## GSM1966868_7851_ExpWBC.CEL                  ftp://ftp.ebi.ac.uk/..
##                            FactorValue..organism.part.     GROUP
##                                            &lt;character&gt; &lt;numeric&gt;
## GSM1966868_7851_ExpWBC.CEL      circulating white bl..         0</code></pre>
<p>Aplicamos un an√°lisis de expresi√≥n diferencial empleando
<strong>limma</strong>.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb57-1" aria-hidden="true" tabindex="-1"></a>se75759 <span class="ot">=</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">deAna</span>(<span class="at">expr =</span> se75759, </span>
<span id="cb57-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb57-2" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">padj.method =</span> <span class="st">"BH"</span>, </span>
<span id="cb57-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb57-3" aria-hidden="true" tabindex="-1"></a>                                   <span class="at">de.method =</span> <span class="st">"limma"</span>)</span>
<span id="cb57-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb57-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">rowData</span>(se75759),<span class="dv">2</span> )</span></code></pre></div>
<pre><code>## DataFrame with 2 rows and 4 columns
##               FC limma.STAT      PVAL  ADJ.PVAL
##        &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## 699677 0.0567034   0.204757  0.840073  0.999891
## 720098 0.0343384   0.407500  0.688475  0.999891</code></pre>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb59-1" aria-hidden="true" tabindex="-1"></a>FDR <span class="ot">=</span> <span class="fl">0.1</span></span>
<span id="cb59-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb59-2" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">rowData</span>(se75759)<span class="sc">$</span>ADJ.PVAL <span class="sc">&lt;</span> FDR )</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<p>Los p valores ajustados que hemos obtenido son muy cercanos a 1, por
tanto, no ser√≠a correcto tomar un valor de FDR tan alto que nos
condujese al rechazo de la hip√≥tesis nula para ninguno de los genes, en
consecuencia, podemos afirmar que no tenemos genes con expresi√≥n
diferencial en funci√≥n del tratamiento.</p>
<p>Los resultados obtenidos coinciden con el T-test de la
<strong>tarea2.3</strong>, utilizando
<em><strong>rowttests</strong></em> del paquete
<em><strong>genefilter</strong></em>.</p>
<p>&nbsp;</p>
<hr>
</div>
<div id="an√°lisis-de-sobre-representaci√≥n." class="section level1">
<h1>5. An√°lisis de sobre representaci√≥n.</h1>
<div id="test-de-fisher" class="section level3">
<h3>Test de Fisher</h3>
<p>A continuaci√≥n realizamos un an√°lisis de sobre representaci√≥n con el
test de Fisher unilateral empleando la funci√≥n <strong>sbea</strong> y
el m√©todo <strong>ora</strong> (overrepresentation analysis).</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb61-1" aria-hidden="true" tabindex="-1"></a>se75759.oraKEGG <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">sbea</span>(<span class="at">method =</span> <span class="st">"ora"</span>, <span class="at">se =</span> se75759,</span>
<span id="cb61-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb61-2" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">padj.method =</span> <span class="st">"BH"</span>,</span>
<span id="cb61-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb61-3" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">gs=</span>mccKEGGgsc, <span class="at">perm =</span> <span class="dv">0</span>,</span>
<span id="cb61-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb61-4" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb61-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb61-5" aria-hidden="true" tabindex="-1"></a><span class="fu">gsRanking</span>(se75759.oraKEGG, <span class="at">signif.only =</span> F)</span></code></pre></div>
<pre><code>## DataFrame with 335 rows and 5 columns
##        GENE.SET  NR.GENES NR.SIG.GENES      PVAL  ADJ.PVAL
##     &lt;character&gt; &lt;numeric&gt;    &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## 1      mcc05200       363            0         1         1
## 2      mcc05022       339            0         1         1
## 3      mcc05168       308            0         1         1
## 4      mcc05010       278            0         1         1
## 5      mcc05014       263            0         1         1
## ...         ...       ...          ...       ...       ...
## 331    mcc00730         7            0         1         1
## 332    mcc04122         7            0         1         1
## 333    mcc00740         6            0         1         1
## 334    mcc00470         5            0         1         1
## 335    mcc00524         5            0         1         1</code></pre>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb63-1" aria-hidden="true" tabindex="-1"></a>se75759.oraGO <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">sbea</span>(<span class="at">method =</span> <span class="st">"ora"</span>, <span class="at">se =</span> se75759, </span>
<span id="cb63-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb63-2" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">padj.method =</span> <span class="st">"BH"</span>,</span>
<span id="cb63-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb63-3" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">gs=</span>mccGOgsc, <span class="at">perm =</span> <span class="dv">0</span>,</span>
<span id="cb63-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb63-4" aria-hidden="true" tabindex="-1"></a>                                         <span class="at">alpha =</span> <span class="fl">0.05</span>)</span>
<span id="cb63-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb63-5" aria-hidden="true" tabindex="-1"></a><span class="fu">gsRanking</span>(se75759.oraGO, <span class="at">signif.only =</span> F)</span></code></pre></div>
<pre><code>## DataFrame with 6592 rows and 5 columns
##         GENE.SET  NR.GENES NR.SIG.GENES      PVAL  ADJ.PVAL
##      &lt;character&gt; &lt;numeric&gt;    &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;
## 1     GO:0023056       497            0         1         1
## 2     GO:0043565       496            0         1         1
## 3     GO:0045184       495            0         1         1
## 4     GO:0010647       494            0         1         1
## 5     GO:0006915       483            0         1         1
## ...          ...       ...          ...       ...       ...
## 6588  GO:2001014         5            0         1         1
## 6589  GO:2001028         5            0         1         1
## 6590  GO:2001046         5            0         1         1
## 6591  GO:2001204         5            0         1         1
## 6592  GO:2001214         5            0         1         1</code></pre>
<p>Como ya hemos mencionado anteriormente, no encontramos expresi√≥n
diferencial de los genes en funci√≥n del tratamiento, y a trav√©s del
an√°lisis <strong>ORA</strong> tampoco obtenemos genes sobrerepresentados
en ninguno de los grupos funcionales definidos por KEGG o GO.</p>
<p>&nbsp;<br>
***</p>
</div>
</div>
<div id="an√°lisis-de-grupos-o-conjuntos-de-genes." class="section level1">
<h1>6. An√°lisis de grupos o conjuntos de genes.</h1>
<p>En este apartado estudiaremos si existe relaci√≥n entre conjuntos de
genes previamente definidos, en este caso tomaremos las colecciones de
genes de Go y KEGG previamente creadas, y el fenotipo.</p>
<p>En este tipo de an√°lisis se formulan las siguiente hip√≥tesis
nulas:</p>
<ul>
<li><p><strong>Q1</strong> : Los genes de un conjunto muestran el mismo
patr√≥n de asociaci√≥n con el fenotipo que los genes pertenecientes al
resto de grupos. Esta hip√≥tesis se centra en la comparaci√≥n entre la
asociaci√≥n de los genes pertenecientes a un conjunto a un fenotipo y la
asociaci√≥n de los genes de otros conjuntos y ese fenotipo. Para esto
haremos un <strong>test competitivo</strong>.</p></li>
<li><p><strong>Q2</strong> : El conjunto de genes no presenta ning√∫n gen
con un nivel de expresi√≥n asociado al fenotipo estudiado. Esta hip√≥tesis
estudia la expresi√≥n diferencial de los genes pertenecientes a cada
conjunto. Para esto haremos un <strong>test
autocontenido</strong>.</p></li>
</ul>
<div id="gsa-para-grupos-kegg" class="section level2">
<h2>GSA para grupos KEGG:</h2>
<p>En este apartado emplearemos <strong>EnrichmentBrowser</strong> para
llevar a cabo un an√°lisis <strong>GSA</strong>. Mediante este test
podemos encontrar los grupos cuyos genes muestran diferencias
estad√≠sticamente significativas entre dos estados o fenotipos, en este
caso el tratamiento con formaldeh√≠do.<br>
Este an√°lisis difiere del GSEA en que nos da una medida del
enriquecimiento (estad√≠sitico maxnean), este toma la media de las partes
positivas (grupos de genes positivos, es decir que en la segunda
condici√≥n han aumentando su expresi√≥n) y la media de las partes
negativas (grupos de genes negativos, es decir que en la segunda
condici√≥n han disminuido su expresi√≥n) y se queda con el m√°ximo de
ambos.<br>
Para el ajuste de los p valores usamos el m√©todo de
<em>Benjamini-Hochberg</em>.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb65-1" aria-hidden="true" tabindex="-1"></a>se75759.gsaKEGG <span class="ot">&lt;-</span> EnrichmentBrowser<span class="sc">::</span><span class="fu">sbea</span>(<span class="at">method =</span> <span class="st">"gsa"</span>, <span class="at">se =</span> se75759,</span>
<span id="cb65-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb65-2" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">padj.method =</span> <span class="st">"BH"</span>,</span>
<span id="cb65-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb65-3" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">gs=</span>mccKEGGgsc,<span class="at">alpha =</span> <span class="fl">0.05</span>)   </span>
<span id="cb65-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb65-4" aria-hidden="true" tabindex="-1"></a><span class="fu">gsRanking</span>(se75759.gsaKEGG, <span class="at">signif.only =</span> F)</span></code></pre></div>
<p>El n√∫mero de grupos significativos encontrados es:</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb66-1" aria-hidden="true" tabindex="-1"></a>se75759.gsaKEGG<span class="sc">$</span>nr.sigs</span></code></pre></div>
<pre><code>## [1] 0</code></pre>
<div id="generaci√≥n-de-informe" class="section level3">
<h3>Generaci√≥n de informe:</h3>
<p>No se han obtenido resultados significativos, aunque con el objetivo
de mostrar los resultados incluiremos todos los grupos en el
informe.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Tomamos los resultados:</span></span>
<span id="cb68-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb68-2" aria-hidden="true" tabindex="-1"></a>tabla <span class="ot">=</span> se75759.gsaKEGG<span class="sc">$</span>res.tbl</span>
<span id="cb68-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb68-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb68-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Creamos la url para cada identificador:</span></span>
<span id="cb68-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb68-5" aria-hidden="true" tabindex="-1"></a>ID <span class="ot">=</span> tabla<span class="sc">$</span>GENE.SET</span>
<span id="cb68-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb68-6" aria-hidden="true" tabindex="-1"></a>KEGGID <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(ID), <span class="cn">NA</span>, <span class="fu">paste0</span>(<span class="st">"&lt;a href='https://www.genome.jp/"</span>,</span>
<span id="cb68-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb68-7" aria-hidden="true" tabindex="-1"></a>                                      <span class="st">"dbget-bin/www_bget?"</span>, ID, <span class="st">"'&gt;"</span>,</span>
<span id="cb68-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb68-8" aria-hidden="true" tabindex="-1"></a>                                      ID, <span class="st">"&lt;/a&gt;"</span>))</span>
<span id="cb68-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb68-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-10"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb68-10" aria-hidden="true" tabindex="-1"></a>final_info_KEGG <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Grupo_KEGG =</span> KEGGID, <span class="at">Score =</span> tabla<span class="sc">$</span>SCORE,</span>
<span id="cb68-11"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb68-11" aria-hidden="true" tabindex="-1"></a>                             <span class="at">P_val =</span> tabla<span class="sc">$</span>PVAL, <span class="at">p_ajust=</span>tabla<span class="sc">$</span>ADJ.PVAL  )  </span></code></pre></div>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb69-1" aria-hidden="true" tabindex="-1"></a>info <span class="ot">=</span> final_info_KEGG</span>
<span id="cb69-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb69-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb69-3" aria-hidden="true" tabindex="-1"></a>f_out <span class="ot">=</span> <span class="st">"T4_GSA_KEGG.DE"</span></span>
<span id="cb69-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb69-4" aria-hidden="true" tabindex="-1"></a>report_dir <span class="ot">=</span> <span class="st">"Reports/"</span></span>
<span id="cb69-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb69-5" aria-hidden="true" tabindex="-1"></a>titulo <span class="ot">=</span> <span class="st">"geod75759 : Resultados GSA de grupos KEGG."</span></span>
<span id="cb69-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb69-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb69-7" aria-hidden="true" tabindex="-1"></a>report1 <span class="ot">=</span> ReportingTools<span class="sc">::</span><span class="fu">HTMLReport</span>(<span class="at">shortName =</span> f_out, </span>
<span id="cb69-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb69-8" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">title =</span> titulo,</span>
<span id="cb69-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb69-9" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">reportDirectory =</span> report_dir)</span>
<span id="cb69-10"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb69-10" aria-hidden="true" tabindex="-1"></a><span class="fu">publish</span>(info,report1)</span>
<span id="cb69-11"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb69-11" aria-hidden="true" tabindex="-1"></a><span class="fu">finish</span>(report1)</span></code></pre></div>
<pre><code>## [1] "Reports//T4_GSA_KEGG.DE.html"</code></pre>
</div>
</div>
<div id="gene-set-analysis-con-tami-para-grupos-go" class="section level2">
<h2>Gene set analysis con tami para grupos Go</h2>
<p>En este apartado emplearemos el paquete <strong>tami</strong> con el
fin de llevar a cabo un an√°lisis de grupos de genes tanto competitivo
como autocontenido usando la funci√≥n <strong>GeneSetTest</strong> con
los siguientes par√°metros:</p>
<ul>
<li><strong>test</strong> -&gt; rowtest<br>
Realizar un an√°lisis de expresi√≥n diferencial marginal (test de la t
moderada de <strong>Limma</strong>).</li>
<li><strong>association</strong> -&gt; ‚Äústatistic‚Äù<br>
Empleamos el estad√≠stico de contraste como asociaci√≥n entre la expresi√≥n
y el fenotipo.<br>
</li>
<li><strong>correction</strong> -&gt; ‚ÄúBH‚Äù<br>
Como m√©todo de ajuste de los p-valores el m√©todo de Benjamini y
Hochberg.<br>
</li>
<li><strong>GeneNullDistr</strong> -&gt; ‚Äúrandomization‚Äù. Contrastamos
la distribuci√≥n nula de aleatorizaci√≥n por columna.<br>
</li>
<li><strong>alternative</strong> -&gt; ‚Äútwo-sided‚Äù. Usamos un test
bilateral.<br>
</li>
<li><strong>nmax</strong> -&gt; ‚Äú100‚Äù. N√∫mero de remuestreos
aleatorios.</li>
</ul>
<div id="hip√≥tesis-nula-competitiva" class="section level3">
<h3>Hip√≥tesis nula competitiva</h3>
<p>La hip√≥tesis nula <strong>Ho</strong> indica que los genes de un
grupo previamente definido S est√°n al menos tan frecuentemente
diferentemente expresados como los genes en el grupo Sc.<br>
Par√°metros:</p>
<ul>
<li><strong>GeneSetNullDistr</strong> = ‚Äúcompetitive‚Äù</li>
</ul>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb71-1" aria-hidden="true" tabindex="-1"></a>geod75759_GO_comp <span class="ot">&lt;-</span> <span class="fu">GeneSetTest</span>(<span class="at">x =</span> geod75759_rma,</span>
<span id="cb71-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb71-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">y=</span><span class="fu">factor</span>(<span class="fu">pData</span>(geod75759_rma)<span class="sc">$</span>Sample_treatment),</span>
<span id="cb71-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb71-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">test =</span> rowtmod, <span class="at">association=</span><span class="st">"statistic"</span>,</span>
<span id="cb71-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb71-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">correction=</span><span class="st">"BH"</span>, <span class="at">GeneNullDistr =</span> <span class="st">"randomization"</span>,</span>
<span id="cb71-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb71-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">GeneSetNullDistr =</span> <span class="st">"competitive"</span>,</span>
<span id="cb71-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb71-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">alternative=</span><span class="st">"two-sided"</span>, <span class="at">nmax =</span> <span class="dv">100</span>,</span>
<span id="cb71-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb71-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">id =</span> <span class="st">"ENTREZID"</span>, <span class="at">gsc=</span><span class="fu">geneIds</span>(mccGOgsc),</span>
<span id="cb71-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb71-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">descriptive=</span>mean, <span class="at">foutput =</span> <span class="st">"geod75759_rma_comp"</span>)</span>
<span id="cb71-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb71-9" aria-hidden="true" tabindex="-1"></a>geod75759_GO_comp_df <span class="ot">=</span> <span class="fu">tidy</span>(geod75759_GO_comp)</span>
<span id="cb71-10"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb71-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-11"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb71-11" aria-hidden="true" tabindex="-1"></a>sig_GO_comp <span class="ot">=</span> geod75759_GO_comp_df<span class="sc">$</span>adjp<span class="sc">&lt;</span><span class="fl">0.05</span></span></code></pre></div>
<p>Tomando una <strong>FDR</strong> de 0.05, ¬øCu√°ntos grupos Go
significativos encontramos en nuestro estudio?</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb72-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(sig_GO_comp)</span></code></pre></div>
<pre><code>## [1] 91</code></pre>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geod75759_GO_comp_df[sig_GO_comp,])</span></code></pre></div>
<pre><code>##                    GO statistic rawp adjp
## GO:0000082 GO:0000082  1.503827    0    0
## GO:0000083 GO:0000083  1.503827    0    0
## GO:0000289 GO:0000289  1.598281    0    0
## GO:0001530 GO:0001530 -1.998616    0    0
## GO:0002062 GO:0002062  1.503827    0    0
## GO:0002761 GO:0002761  1.503827    0    0</code></pre>
</div>
<div id="generaci√≥n-de-informe-1" class="section level3">
<h3>Generaci√≥n de informe:</h3>
<p>Mostrasmos en un informe los grupos significativos tonando una FDR de
0.05.</p>
<div class="sourceCode" id="cb76"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb76-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb76-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Tomamos los resultados:</span></span>
<span id="cb76-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb76-2" aria-hidden="true" tabindex="-1"></a>tabla2 <span class="ot">=</span> geod75759_GO_comp_df[sig_GO_comp,]</span>
<span id="cb76-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb76-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb76-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Creamos la url para cada identificador:</span></span>
<span id="cb76-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb76-5" aria-hidden="true" tabindex="-1"></a>ID <span class="ot">=</span> tabla2<span class="sc">$</span>GO</span>
<span id="cb76-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb76-6" aria-hidden="true" tabindex="-1"></a>GOID <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(ID), <span class="cn">NA</span>, tami<span class="sc">::</span><span class="fu">go2url</span>(ID))</span>
<span id="cb76-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb76-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb76-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb76-8" aria-hidden="true" tabindex="-1"></a>final_comp_GO <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Grupo_GO =</span> GOID, <span class="at">satistic =</span> tabla2<span class="sc">$</span>statistic,</span>
<span id="cb76-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb76-9" aria-hidden="true" tabindex="-1"></a>                             <span class="at">p_value =</span> tabla2<span class="sc">$</span>rawp, <span class="at">p_ajust=</span>tabla2<span class="sc">$</span>adjp)  </span></code></pre></div>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb77-1" aria-hidden="true" tabindex="-1"></a>info <span class="ot">=</span> final_comp_GO</span>
<span id="cb77-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb77-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb77-3" aria-hidden="true" tabindex="-1"></a>f_out <span class="ot">=</span> <span class="st">"T4_GSA_comp_GO.DE"</span></span>
<span id="cb77-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb77-4" aria-hidden="true" tabindex="-1"></a>report_dir <span class="ot">=</span> <span class="st">"Reports/"</span></span>
<span id="cb77-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb77-5" aria-hidden="true" tabindex="-1"></a>titulo <span class="ot">=</span> <span class="st">"geod75759 : Resultados GSA de grupos Go (hipotesis competitiva). "</span></span>
<span id="cb77-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb77-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb77-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb77-7" aria-hidden="true" tabindex="-1"></a>reportGO1 <span class="ot">=</span> ReportingTools<span class="sc">::</span><span class="fu">HTMLReport</span>(<span class="at">shortName =</span> f_out, </span>
<span id="cb77-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb77-8" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">title =</span> titulo,</span>
<span id="cb77-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb77-9" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">reportDirectory =</span> report_dir)</span>
<span id="cb77-10"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb77-10" aria-hidden="true" tabindex="-1"></a><span class="fu">publish</span>(info,reportGO1)</span>
<span id="cb77-11"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb77-11" aria-hidden="true" tabindex="-1"></a><span class="fu">finish</span>(reportGO1)</span></code></pre></div>
<pre><code>## [1] "Reports//T4_GSA_comp_GO.DE.html"</code></pre>
</div>
<div id="hip√≥tesis-nula-autocontenida" class="section level3">
<h3>Hip√≥tesis nula autocontenida</h3>
<p>La hip√≥tesis nula mantiene que ning√∫n gen perteneciente a un grupo S
est√° diferencialmente expresado.<br>
Par√°metros:</p>
<ul>
<li><strong>GeneSetNullDistr</strong> = ‚Äúself-contained‚Äù</li>
</ul>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb79-1" aria-hidden="true" tabindex="-1"></a>geod75759_GO_self <span class="ot">&lt;-</span> <span class="fu">GeneSetTest</span>(<span class="at">x =</span> geod75759_rma,</span>
<span id="cb79-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb79-2" aria-hidden="true" tabindex="-1"></a>                              <span class="at">y=</span><span class="fu">factor</span>(<span class="fu">pData</span>(geod75759_rma)<span class="sc">$</span>Sample_treatment),</span>
<span id="cb79-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb79-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">test =</span> rowtmod, <span class="at">association=</span><span class="st">"statistic"</span>,</span>
<span id="cb79-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb79-4" aria-hidden="true" tabindex="-1"></a>                              <span class="at">correction=</span><span class="st">"BH"</span>, <span class="at">GeneNullDistr =</span> <span class="st">"randomization"</span>,</span>
<span id="cb79-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb79-5" aria-hidden="true" tabindex="-1"></a>                              <span class="at">GeneSetNullDistr =</span> <span class="st">"self-contained"</span>,</span>
<span id="cb79-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb79-6" aria-hidden="true" tabindex="-1"></a>                              <span class="at">alternative=</span><span class="st">"two-sided"</span>, <span class="at">nmax =</span> <span class="dv">100</span>,</span>
<span id="cb79-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb79-7" aria-hidden="true" tabindex="-1"></a>                              <span class="at">id =</span> <span class="st">"ENTREZID"</span>, <span class="at">gsc=</span><span class="fu">geneIds</span>(mccGOgsc),</span>
<span id="cb79-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb79-8" aria-hidden="true" tabindex="-1"></a>                              <span class="at">descriptive=</span>mean, <span class="at">foutput =</span> <span class="st">"geod75759_rma_self"</span>)</span>
<span id="cb79-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb79-9" aria-hidden="true" tabindex="-1"></a>geod75759_GO_self_df <span class="ot">=</span> <span class="fu">tidy</span>(geod75759_GO_self)</span>
<span id="cb79-10"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb79-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-11"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb79-11" aria-hidden="true" tabindex="-1"></a>sig_GO_self <span class="ot">=</span> geod75759_GO_self_df<span class="sc">$</span>adjp<span class="sc">&lt;</span><span class="fl">0.05</span></span></code></pre></div>
<p>Tomando una <strong>FDR</strong> de 0.05, ¬øCu√°ntos grupos Go
significativos encontramos en nuestro estudio?</p>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb80-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(sig_GO_self)</span></code></pre></div>
<pre><code>## [1] 40</code></pre>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(geod75759_GO_self_df[sig_GO_self,])</span></code></pre></div>
<pre><code>##                    GO statistic rawp adjp
## GO:0001530 GO:0001530 -1.998616    0    0
## GO:0002862 GO:0002862 -1.998616    0    0
## GO:0005227 GO:0005227  1.751634    0    0
## GO:0005548 GO:0005548 -1.979465    0    0
## GO:0005697 GO:0005697 -2.629894    0    0
## GO:0006997 GO:0006997 -2.629894    0    0</code></pre>
</div>
<div id="generaci√≥n-de-informe-2" class="section level3">
<h3>Generaci√≥n de informe:</h3>
<p>Usamos los grupos de genes significativos tomando una
<strong>FDR</strong> de 0.05:</p>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb84-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Tomamos los resultados:</span></span>
<span id="cb84-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb84-2" aria-hidden="true" tabindex="-1"></a>tabla3 <span class="ot">=</span> geod75759_GO_self_df[sig_GO_self,]</span>
<span id="cb84-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb84-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb84-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Creamos la url para cada identificador:</span></span>
<span id="cb84-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb84-5" aria-hidden="true" tabindex="-1"></a>ID <span class="ot">=</span> tabla3<span class="sc">$</span>GO</span>
<span id="cb84-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb84-6" aria-hidden="true" tabindex="-1"></a>GOID <span class="ot">=</span> <span class="fu">ifelse</span>(<span class="fu">is.na</span>(ID), <span class="cn">NA</span>, tami<span class="sc">::</span><span class="fu">go2url</span>(ID))</span>
<span id="cb84-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb84-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb84-8" aria-hidden="true" tabindex="-1"></a>final_self_GO <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="at">Grupo_GO =</span> GOID, <span class="at">satistic =</span> tabla3<span class="sc">$</span>statistic,</span>
<span id="cb84-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb84-9" aria-hidden="true" tabindex="-1"></a>                            <span class="at">p_value =</span> tabla3<span class="sc">$</span>rawp, <span class="at">p_ajust=</span>tabla3<span class="sc">$</span>adjp)  </span></code></pre></div>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb85-1" aria-hidden="true" tabindex="-1"></a>info <span class="ot">=</span> final_self_GO</span>
<span id="cb85-2"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb85-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-3"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb85-3" aria-hidden="true" tabindex="-1"></a>f_out <span class="ot">=</span> <span class="st">"T4_GSA_self_GO.DE"</span></span>
<span id="cb85-4"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb85-4" aria-hidden="true" tabindex="-1"></a>report_dir <span class="ot">=</span> <span class="st">"Reports/"</span></span>
<span id="cb85-5"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb85-5" aria-hidden="true" tabindex="-1"></a>titulo <span class="ot">=</span> <span class="st">"geod75759 : Resultados GSA de grupos Go (hipotesis autocomparativa). "</span></span>
<span id="cb85-6"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb85-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb85-7"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb85-7" aria-hidden="true" tabindex="-1"></a>reportGO2 <span class="ot">=</span> ReportingTools<span class="sc">::</span><span class="fu">HTMLReport</span>(<span class="at">shortName =</span> f_out, </span>
<span id="cb85-8"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb85-8" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">title =</span> titulo,</span>
<span id="cb85-9"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb85-9" aria-hidden="true" tabindex="-1"></a>                                      <span class="at">reportDirectory =</span> report_dir)</span>
<span id="cb85-10"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb85-10" aria-hidden="true" tabindex="-1"></a><span class="fu">publish</span>(info,reportGO2)</span>
<span id="cb85-11"><a href="file:///C:/Users/roxya/OneDrive/Documentos/Paquete_R/PacRoxanaMoldovan/vignettes/tarea4.html#cb85-11" aria-hidden="true" tabindex="-1"></a><span class="fu">finish</span>(reportGO2)</span></code></pre></div>
<pre><code>## [1] "Reports//T4_GSA_self_GO.DE.html"</code></pre>
<hr>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>



</body></html>