<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Roxana Andreea Moldovan Moldovan" />

<meta name="date" content="2022-06-11" />

<title>Tarea3. PRJNA821482: Construyendo un SummarizedExperiment</title>

<script>// Pandoc 2.9 adds attributes on both header and div. We remove the former (to
// be compatible with the behavior of Pandoc < 2.8).
document.addEventListener('DOMContentLoaded', function(e) {
  var hs = document.querySelectorAll("div.section[class*='level'] > :first-child");
  var i, h, a;
  for (i = 0; i < hs.length; i++) {
    h = hs[i];
    if (!/^h[1-6]$/i.test(h.tagName)) continue;  // it should be a header h1-h6
    a = h.attributes;
    while (a.length > 0) h.removeAttribute(a[0].name);
  }
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>




<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">Tarea3. PRJNA821482: Construyendo un
SummarizedExperiment</h1>
<h4 class="author">Roxana Andreea Moldovan Moldovan</h4>
<h4 class="date">2022-06-11</h4>



<hr />
<div id="introducción" class="section level1">
<h1>1. Introducción</h1>
<p>En esta tarea construiremos un SummarizedExperiment a partir de un
estudio de RNA-seq, incluyendo la descarga con
<strong>SRA-Toolkit</strong>, el alineamiento con
<strong>bowtie2</strong> y la inclusión de toda la información
disponible en el SummarizedExperiment.</p>
<div id="datos-del-exprimento" class="section level2">
<h2>Datos del exprimento:</h2>
<p>Experimento: <a href="https://www.ncbi.nlm.nih.gov/bioproject/821482">PRJNA821482</a>.</p>
</div>
<div id="información-relevante" class="section level2">
<h2>Información relevante:</h2>
<div id="hypoxia-promotes-erythroid-differentiation-human" class="section level3">
<h3>Hypoxia promotes erythroid differentiation (human)</h3>
<p>En esta estudio se pretende estudiar el efecto de la hipoxia
(crecimiento en ausencia de oxigeno) de las células madre y progenitores
hematopoyéticos (HSPC) en humanos. Para ello se toman muestras de dos
líneas celulares <strong>HSC</strong> y <strong>K562</strong>, que se
corresponden a células madre hematopoyéticas y linfoblastos inmortales
obtenidas de un paciente de 53 años con leucemia mieloide crónica. Ambos
tipos celulares se han cultivado en presencie y ausencia de oxígeno
(<strong>Normoxia</strong> e <strong>Hipoxia</strong>). Para cada
experimento se han realizado 3 repeticiones obteniendo así 12 muestras
que se han secuneciado con <strong>Illumina NovaSeq 6000</strong>.</p>
<p> </p>
<hr />
</div>
</div>
</div>
<div id="obtención-de-datos" class="section level1">
<h1>2. Obtención de datos</h1>
<div id="descaraga-de-datos" class="section level2">
<h2>Descaraga de datos:</h2>
<p>Vamos al SRA Run Selector, donde encontramos la información del
experimento. El botón <em><strong>Accesion List</strong></em> nos
permite guardar los nombres de los datos que emplearemos en un fichero
llamado <em>SRR_Acc_List.txt</em>.</p>
<p>Mientras que <em><strong>Metadata</strong></em> nos permite guardar
las variables fenotípicas y covariables de cada muestra en el fichero
<em>SraRunTable.txt</em>.</p>
<p>Siempre trabajamos desde el mismo directorio.</p>
<p>Para descargar los datos empelaremos <strong>prefetch</strong> de
<strong>SRA-Toolkit</strong>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Creamos el directorio que contendrá los datos: </span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> PRJNA821482</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mkdir</span> PRJNA821482/sra</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Descargamos los datos en el directorio PRJNA821482/sra</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="ex">prefetch</span> <span class="at">--option-file</span> SRR_Acc_List.txt <span class="at">--output-directory</span> PRJNA821482/sra</span></code></pre></div>
<p>Tenemos la información contenida en ficheros SRA, debemos pasar estos
a FASTQ, para ello usaremos la opción <strong>fasta-dump</strong> de
<strong>SRA-Toolkit</strong>.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Trabajamos desde el mismo directorio, outdir me dice donde guardar los fastq:</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Convertimos los fasta a fastq:</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span> SRR_Acc_List.txt <span class="kw">|</span> <span class="ex">parallel</span> <span class="at">-j</span> 4 fastq-dump <span class="at">--split-files</span> PRJNA821482/sra/{}/{}.sra <span class="at">--outdir</span> PRJNA821482/</span></code></pre></div>
<div id="eliminamos-los-.sra" class="section level3">
<h3>Eliminamos los .sra</h3>
<p>Una vez llevada a cabo la conversión de <em>.sra</em> a
<em>.fastq</em> eliminaremos el directorio
<em><strong>PRJNA821482/sra</strong></em> con el fin de liberar espacio,
mientras que mantenemos los fastq en el directorio
<em><strong>PRJNA821482</strong></em> .</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Eliminamos el directorio que contenía los ficheros sra:</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> <span class="at">-r</span> PRJNA821482/sra</span></code></pre></div>
<p> </p>
<hr />
</div>
</div>
</div>
<div id="alineamiento" class="section level1">
<h1>3. Alineamiento</h1>
<p>Para llevar a cabo el alineamiento vamos a usar
<strong>Bowtie2</strong>, un software de alineamiento sobre un genoma de
referencia.</p>
<div id="descargamos-el-genoma-de-referencia" class="section level2">
<h2>Descargamos el genoma de referencia</h2>
<p>El organismo de estudio es Homo sapiens[Taxonomy ID: 9606].
Descargamos su genoma de referencia más reciente (GRCh38) partir de la
web de <a href="https://support.illumina.com/sequencing/sequencing_software/igenome.html">Ilumina</a>
y la almacenaremos en el directorio
<strong>PRJNA821482/referencia</strong>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Descargamos el genoma de referencia en el directorio PRJNA821482/referencia</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> http://igenomes.illumina.com.s3-website-us-east-1.amazonaws.com/Homo_sapiens/NCBI/GRCh38/Homo_sapiens_NCBI_GRCh38.tar.gz <span class="at">-P</span> PRJNA821482/referencia</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Descomprimimos la información descargada:</span></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># los resultados descomprimidos seguirán en el directorio PRJNA821482/referencia</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a><span class="fu">tar</span> <span class="at">-xvzf</span> PRJNA821482/referencia/Homo_sapiens_NCBI_GRCh38.tar.gz <span class="at">--directory</span> PRJNA821482/referencia</span></code></pre></div>
</div>
<div id="alineamiento-1" class="section level2">
<h2>Alineamiento</h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Fijamos la ruta en la que encontramos el genoma de referencia</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="va">ruta_genoma</span><span class="op">=</span>PRJNA821482/referencia/Homo_sapiens/NCBI/GRCh38/Sequence/Bowtie2Index</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="va">bt2_idx</span><span class="op">=</span><span class="st">&quot;</span><span class="va">$ruta_genoma</span><span class="st">&quot;</span>/genome</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Los ficheros fastq tiene el nombre del identificador seguido de _1 o _2 .fastq</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Listamos el la ruta de los ficheros fastq_1 a alinear:</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> PRJNA821482/<span class="pp">*</span>_1.fastq <span class="op">&gt;</span> lista_fastq.txt</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Tomamos el nombre de los ficheros sin extensión y los alineamos.</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Los ficheros se guardaran en el mismo directorio que los fastq (PRJNA821482)</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot;_&quot;</span> <span class="at">-f</span> 1 lista_fastq.txt <span class="kw">|</span> <span class="ex">parallel</span> <span class="at">-j</span> 10 bowtie2 <span class="at">-x</span> <span class="va">$bt2_idx</span> <span class="at">-1</span> {}_1.fastq <span class="at">-2</span> {}_2.fastq <span class="at">-S</span> {}.sam</span></code></pre></div>
</div>
<div id="conversión-de-sam-a-bam" class="section level2">
<h2>Conversión de SAM a BAM</h2>
<p>Con ayuda de <strong>SamTools</strong> llevamos a cabo la conversión
de los fichero .bam obtenidos con <strong>bowtie2</strong> a archivos
.sam</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Guardamos el nombre de los ficheros sam del directorio PRJNA821482</span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> PRJNA821482/<span class="pp">*</span>.sam <span class="op">&gt;</span> lista_sam.txt</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Listamos el nombre de los ficheros sin extension y convertimos de sam a bam:</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot;.&quot;</span> <span class="at">-f</span> 1 lista_sam.txt <span class="kw">|</span> <span class="ex">parallel</span> <span class="at">-j</span> 2 samtools view {}.sam <span class="at">-S</span> <span class="at">-b</span> <span class="at">-o</span> {}.bam <span class="at">--threads</span> 4</span></code></pre></div>
<div id="eliminamos-.fastq-y-.sam" class="section level3">
<h3>Eliminamos .fastq y .sam</h3>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Eliminamos los dicheros .fastq y .sam para liberar espacio</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> PRJNA821482/<span class="pp">*</span>.fastq</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span> PRJNA821482/<span class="pp">*</span>.sam</span></code></pre></div>
</div>
<div id="ordenamos-las-lecturas" class="section level3">
<h3>Ordenamos las lecturas</h3>
<p>Usamos la opción <em><strong>sort</strong></em> de
<strong>samtools</strong> para ordenar las lecturas.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Tomamos el nombre de los ficheros y ordenamos</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="fu">cut</span> <span class="at">-d</span> <span class="st">&quot;.&quot;</span> <span class="at">-f</span> 1 lista_sam.txt <span class="kw">|</span> <span class="ex">parallel</span> <span class="at">-j</span> 2 samtools sort {}.bam <span class="at">-o</span> {}_sort.bam <span class="at">--threads</span>  4</span></code></pre></div>
<p> </p>
<hr />
</div>
</div>
</div>
<div id="conteo" class="section level1">
<h1>4. Conteo</h1>
<p>Con ayuda de <strong>Rsamtools</strong>, a partir de la anotación de
los genes y los resultados obtenidos tras el alineamiento para cada una
de las muestras, podemos construir la matriz de conteos.</p>
<div id="paquetes-a-cargar" class="section level2">
<h2>Paquetes A cargar</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Carga de paquetes:</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(Rsamtools,GenomicFeatures, GenomicAlignments,S4Vectors)</span></code></pre></div>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Listamos los fichero sort.bam con su ruta:</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>indir <span class="ot">=</span> <span class="fu">getwd</span>()</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>fich_bam <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> <span class="fu">paste0</span>(indir, <span class="st">&quot;/PRJNA821482&quot;</span>),</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>                       <span class="at">pattern =</span> <span class="st">&quot;*.sort.bam&quot;</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a><span class="co">#Creamos una referencia a estos:</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>bamLst <span class="ot">=</span> Rsamtools<span class="sc">::</span><span class="fu">BamFileList</span>(fich_bam, <span class="at">index=</span><span class="fu">character</span>(),</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>                                <span class="at">yieldSize=</span><span class="dv">100000</span>, <span class="at">obeyQname =</span> <span class="cn">TRUE</span>)</span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Fijamos la ruta del fichero genes.gtf</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>ruta_gtf<span class="ot">=</span> <span class="fu">paste0</span>(indir,<span class="st">&quot;/PRJNA821482/referencia/Homo_sapiens/NCBI/GRCh38&quot;</span>,</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>                 <span class="st">&quot;/Annotation/Archives/archive-2015-08-11-09-31-31&quot;</span>,</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>                 <span class="st">&quot;/Genes/genes.gtf&quot;</span>)</span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Generamos un objeto de tipo TxDb a partir de estos:</span></span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">=</span> <span class="fu">makeTxDbFromGFF</span>(ruta_gtf, <span class="at">format=</span><span class="st">&quot;gtf&quot;</span>, <span class="at">organism =</span> <span class="st">&quot;Homo sapiens&quot;</span>)</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Obtenemos los datos de anotación:</span></span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>genes <span class="ot">=</span> <span class="fu">exonsBy</span>(txdb, <span class="at">by=</span><span class="st">&quot;gene&quot;</span>)</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Relacionamos las lecturas con las características genómicas:</span></span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>PRJNA821482 <span class="ot">&lt;-</span> <span class="fu">summarizeOverlaps</span>(<span class="at">features =</span> genes, </span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">read =</span> bamLst,</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">mode=</span> <span class="st">&quot;Union&quot;</span>,</span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">singleEnd=</span><span class="cn">FALSE</span>, <span class="co"># Usa pair-ends</span></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">ignore.strand=</span><span class="cn">TRUE</span>,</span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">fragments=</span><span class="cn">FALSE</span>)</span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(PRJNA821482, <span class="at">file=</span>(<span class="fu">paste0</span>(dir_datos,<span class="st">&quot;PRJNA821482_brutos.rda&quot;</span>)))</span></code></pre></div>
</div>
<div id="información-del-experimento" class="section level2">
<h2>Información del experimento</h2>
<p>Nos encontramos ante un Ranged Summarized Experiment, esto es
importante ya que posteriormente usaremos el paquete
<strong>SummarizedExperiment</strong> para trabajar con los datos.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(SummarizedExperiment, GenomicRanges)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a><span class="fu">class</span>(PRJNA821482)</span></code></pre></div>
<pre><code>## [1] &quot;RangedSummarizedExperiment&quot;
## attr(,&quot;package&quot;)
## [1] &quot;SummarizedExperiment&quot;</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(PRJNA821482)</span></code></pre></div>
<pre><code>## [1] 33121    12</code></pre>
<p>Disponemos de 33121 genes y 12 muestras.</p>
<p>Tenemos una matriz de conteo cuyas columnas se corresponden con
muestras y cuyas filas son los genes.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">assay</span>(PRJNA821482),<span class="dv">2</span>)</span></code></pre></div>
<pre><code>##          SRR18537220_sort.bam SRR18537221_sort.bam SRR18537222_sort.bam
## A1BG                       81                  131                   74
## A1BG-AS1                    9                   24                   12
##          SRR18537223_sort.bam SRR18537224_sort.bam SRR18537225_sort.bam
## A1BG                       83                   72                   90
## A1BG-AS1                   14                   14                   12
##          SRR18537226_sort.bam SRR18537227_sort.bam SRR18537228_sort.bam
## A1BG                       43                   82                   44
## A1BG-AS1                   10                    7                   14
##          SRR18537229_sort.bam SRR18537230_sort.bam SRR18537231_sort.bam
## A1BG                       76                   64                   49
## A1BG-AS1                   13                   10                   15</code></pre>
<div id="summarizedexperiment" class="section level3">
<h3>SummarizedExperiment</h3>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Obtenemos los metadatos:</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>metadat <span class="ot">&lt;-</span> <span class="fu">read.table</span>( <span class="fu">paste0</span>(dir_datos, <span class="st">&quot;SummarizedExperiment/SraRunTable.txt&quot;</span>),</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>                      <span class="at">sep =</span> <span class="st">&quot;,&quot;</span>, <span class="at">header =</span> T)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Creamos las variables de tipo factor</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>TipoCelular <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">levels =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>,</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;HSC&quot;</span>, <span class="st">&quot;K562&quot;</span>))</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>Tratamiento <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">0</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>), <span class="at">levels =</span> <span class="dv">0</span><span class="sc">:</span><span class="dv">1</span>,</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>                      <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Hypoxia&quot;</span>, <span class="st">&quot;Normoxia&quot;</span>))</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Añadimos la información a colData</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">DataFrame</span>(<span class="at">Run=</span>metadat<span class="sc">$</span>Run, <span class="at">Muestra =</span> metadat<span class="sc">$</span>Library.Name,</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">TipoCelular=</span>TipoCelular, <span class="at">Tratamiento=</span>Tratamiento,  </span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">BioProject =</span> metadat<span class="sc">$</span>BioProject )</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(PRJNA821482, <span class="at">use.names=</span>T) <span class="ot">=</span> d</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Asignamos estos nombres a las columnas</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(PRJNA821482) <span class="ot">&lt;-</span> <span class="fu">colData</span>(PRJNA821482)<span class="sc">$</span>Run</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">colData</span>(PRJNA821482), <span class="dv">1</span>)</span></code></pre></div>
<pre><code>## DataFrame with 1 row and 5 columns
##                     Run     Muestra TipoCelular Tratamiento  BioProject
##             &lt;character&gt; &lt;character&gt;    &lt;factor&gt;    &lt;factor&gt; &lt;character&gt;
## SRR18537220 SRR18537220  GSM5984186         HSC     Hypoxia PRJNA821482</code></pre>
</div>
<div id="datos-del-experimento" class="section level3">
<h3>Datos del experimento</h3>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>info_ex <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">name =</span> <span class="st">&quot;PRJNA821482&quot;</span>, <span class="at">organism =</span> <span class="st">&quot;Homo sapiens&quot;</span>,</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>                <span class="at">type =</span> <span class="st">&quot;Expression profiling by high throughput sequencing&quot;</span>,</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>                <span class="at">lab =</span> <span class="st">&quot;Beijing Institute of Radiation Medicine&quot;</span>,</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">contact =</span><span class="st">&quot;quanc1989@gmail.com&quot;</span>,</span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>                <span class="at">title =</span> <span class="fu">paste0</span>(<span class="st">&quot;Hypoxia promotes erythroid differentiation&quot;</span>,</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; throughautophagy induced by the HIF/REDD1/mTORC1 signaling&quot;</span>,</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; axis&quot;</span>),</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>                <span class="at">abstract =</span> <span class="fu">paste0</span>(<span class="st">&quot;For hematopoietic stem and progenitor cells&quot;</span>,</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; (HSPCs), hypoxia is a specific microenvironment known as the&quot;</span>,</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; hypoxic niche. The role of hypoxia in maintaining quiescence,&quot;</span>,</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; self-renewal and proliferation of HSPCs has been more&quot;</span>,</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; elaborated. What remains unclear is   the regulation of&quot;</span>,</span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; hypoxia on HSPC differentiation, especially on the&quot;</span>,</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; differentiation of HSPCs into erythrocytes. In this study,&quot;</span>,</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; we show that hypoxia evidently accelerates erythroid&quot;</span>,</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; differentiation, and in this process effectively enhances&quot;</span>,</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; autophagy which rapidly supplies cellular components through&quot;</span>,</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; degradation of organelles or proteins to meet the needs of&quot;</span>,</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; cell stress or development. In order to ascertain whether&quot;</span>,</span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; autophagy mediates the erythroid differentiation promoted&quot;</span>,</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; by hypoxia, we analyze the perturbation of erythroid&quot;</span>,</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; differentiation after pharmacological and genetic&quot;</span>,</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; interference with autophagy, and address that autophagy is&quot;</span>,</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; required for hypoxia-accelerated erythroid differentiation.&quot;</span>,</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; Transcriptomics reveals HIF-1 and mTORC1 pathways may&quot;</span>,</span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; function as the upstream of autophagy to regulate erythroid&quot;</span>,</span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; differentiation under hypoxia. We further determine that&quot;</span>,</span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; mTORC1 signaling is suppressed by hypoxia to relieve its&quot;</span>,</span>
<span id="cb21-29"><a href="#cb21-29" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; inhibition of autophagy. In addition, we also discover a&quot;</span>,</span>
<span id="cb21-30"><a href="#cb21-30" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; new regulatory pattern that with the process of erythroid&quot;</span>,</span>
<span id="cb21-31"><a href="#cb21-31" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; differentiation, mTORC1 activity gradually decreases and&quot;</span>,</span>
<span id="cb21-32"><a href="#cb21-32" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; autophagy activity increases concomitantly. Moreover, we&quot;</span>,</span>
<span id="cb21-33"><a href="#cb21-33" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; provide evidence that HIF-1 target gene REDD1, which&quot;</span>,</span>
<span id="cb21-34"><a href="#cb21-34" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; inhibits mTORC1 activity, is dramatically upregulated under&quot;</span>,</span>
<span id="cb21-35"><a href="#cb21-35" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; hypoxia. Silencing REDD1 expression activates mTORC1&quot;</span>,</span>
<span id="cb21-36"><a href="#cb21-36" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; signaling and impairs the enhanced autophagy and the&quot;</span>,</span>
<span id="cb21-37"><a href="#cb21-37" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; accelerated erythroid differentiation under hypoxia.&quot;</span>,</span>
<span id="cb21-38"><a href="#cb21-38" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; Together, our study reveals that hypoxia is conductive to&quot;</span>,</span>
<span id="cb21-39"><a href="#cb21-39" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; accelerating erythroid differentiation, in which autophagy&quot;</span>,</span>
<span id="cb21-40"><a href="#cb21-40" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; induced by the HIF-1/REDD1/mTORC1 pathway plays a pivotal&quot;</span>,</span>
<span id="cb21-41"><a href="#cb21-41" aria-hidden="true" tabindex="-1"></a>                <span class="st">&quot; role.&quot;</span>),</span>
<span id="cb21-42"><a href="#cb21-42" aria-hidden="true" tabindex="-1"></a>                <span class="at">url=</span><span class="st">&quot;https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE199778&quot;</span>)</span>
<span id="cb21-43"><a href="#cb21-43" aria-hidden="true" tabindex="-1"></a><span class="fu">metadata</span>(PRJNA821482) <span class="ot">&lt;-</span> info_ex</span></code></pre></div>
</div>
<div id="anotación" class="section level3">
<h3>Anotación</h3>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>pacman<span class="sc">::</span><span class="fu">p_load</span>(AnnotationDbi,org.Hs.eg.db)</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Tomamos otros identificadores</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(PRJNA821482)</span></code></pre></div>
<pre><code>## DataFrame with 33121 rows and 0 columns</code></pre>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a>anot <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(org.Hs.eg.db, <span class="at">keys =</span> <span class="fu">rownames</span>(PRJNA821482),</span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">&quot;ENTREZID&quot;</span>,<span class="st">&quot;ENSEMBL&quot;</span>),</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>                              <span class="at">keytype =</span> <span class="st">&quot;ALIAS&quot;</span>)</span></code></pre></div>
<pre><code>## &#39;select()&#39; returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Eliminamos las sondas y genes repetidos:</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>a <span class="ot">&lt;-</span> <span class="fu">match</span>(<span class="fu">unique</span>(anot<span class="sc">$</span>ENTREZID),anot<span class="sc">$</span>ENTREZID)</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>datos_anot1 <span class="ot">&lt;-</span> anot[a,]</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>b <span class="ot">&lt;-</span> <span class="fu">match</span>(<span class="fu">unique</span>(datos_anot1<span class="sc">$</span>ALIAS),datos_anot1<span class="sc">$</span>ALIAS)</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>datos_anot2 <span class="ot">&lt;-</span> datos_anot1[b,]</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Hacemos la correspondencia entre los identificadores y las columnas</span></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>indice <span class="ot">=</span> <span class="fu">match</span>(<span class="fu">rownames</span>(PRJNA821482), datos_anot2<span class="sc">$</span>ALIAS)</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a><span class="fu">rowData</span>(PRJNA821482) <span class="ot">=</span> datos_anot2[indice,]</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Asignamos los datos y eliminamos las sondas que no se corresponden con genes:</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>PRJNA821482 <span class="ot">&lt;-</span> PRJNA821482[<span class="fu">which</span>(<span class="sc">!</span><span class="fu">is.na</span>(<span class="fu">rowData</span>(PRJNA821482)<span class="sc">$</span>ENTREZID)),]</span></code></pre></div>
<p>La información de la que disponemos en el SumarizedExperiment es:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>(PRJNA821482)</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 25189 12 
## metadata(8): name organism ... abstract url
## assays(1): counts
## rownames(25189): A1BG A1BG-AS1 ... ZZEF1 ZZZ3
## rowData names(3): ALIAS ENTREZID ENSEMBL
## colnames(12): SRR18537220 SRR18537221 ... SRR18537230 SRR18537231
## colData names(5): Run Muestra TipoCelular Tratamiento BioProject</code></pre>
</div>
<div id="guardado" class="section level3">
<h3>Guardado</h3>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save</span>(PRJNA821482, <span class="at">file=</span>(<span class="fu">paste0</span>(dir_datos,<span class="st">&quot;PRJNA821482.rda&quot;</span>)))</span></code></pre></div>
<p> </p>
<hr />
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
